Content.Shared/Materials/SharedMaterialReclaimerSystem.cs

```diff
@@ -35,10 +40,26 @@ public override void Initialize()
         SubscribeLocalEvent<MaterialReclaimerComponent, EntityUnpausedEvent>(OnUnpaused);
         SubscribeLocalEvent<MaterialReclaimerComponent, ExaminedEvent>(OnExamined);
         SubscribeLocalEvent<MaterialReclaimerComponent, GotEmaggedEvent>(OnEmagged);
+        SubscribeLocalEvent<MaterialReclaimerComponent, AttemptDamageContactEvent>(OnAttemptDamageContact);
         SubscribeLocalEvent<MaterialReclaimerComponent, MapInitEvent>(OnMapInit);
         SubscribeLocalEvent<CollideMaterialReclaimerComponent, StartCollideEvent>(OnCollide);
         SubscribeLocalEvent<ActiveMaterialReclaimerComponent, ComponentStartup>(OnActiveStartup);
         SubscribeLocalEvent<ActiveMaterialReclaimerComponent, EntityUnpausedEvent>(OnActiveUnpaused);
+        SubscribeLocalEvent<MobStateChangedEvent>(OnMobStateChanged);
+    }
+
+    private void OnMobStateChanged(MobStateChangedEvent ev)
+    {
+        if (ev.NewMobState != MobState.Dead)
+            return;
+        TryComp<MaterialReclaimerComponent>(ev.Origin, out var reclaimer);
+        foreach (var entity in _entityLookupSystem.GetEntitiesInRange(ev.Target, 0.4f, LookupFlags.Approximate | LookupFlags.Static))
+        {
+            if (HasComp<MaterialReclaimerComponent>(entity))
+            {
+                TryStartProcessItem(entity, ev.Target, reclaimer);
+            }
+        }
```

Comment:
this is a really expensive check to be running every time any mob dies. doesn't damagecontacts add a component to a mob when in collision? It should just use that component.

Reply:
I assume you mean check for `DamagedByContactComponent` when a mob dies, and return if not found? I'm still new to larger coding projects, so my view for expense is skewed.

Reply:
You shouldn't be subscribing globally to every mob state change, just use whatever relevant component / event for recycler instead of listening to everything.